### ISSUE TITLE
Add UNITID_<pkgname> to cabal_macros.h

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#7027]
I don't see a drawbacks in that. @RyanGlScott is worried that it might cause a lot more compilation, but I think unitids are not much more volatile than "the set of versions of direct dependencies".

**Comment by RyanGlScott** [comment]
Indeed, `cabal_macros.h` will only change today if the version numbers of one of your direct dependencies changes. If unit IDs were added to `cabal_macros.h`, then it would change whenever _any_ dependency, direct or indirect, changes. But as you say, I'm unclear on how much more common the latter situation is than the former.

**Comment by jneira** [comment]
hi, what would be the motivation to add it? maybe it is obvious but I think it worths to note it explicitly, thanks!

**Comment by RyanGlScott** [comment]
I'll be honest—I've forgotten myself what the context of the issue was. Perhaps @phadej remembers?

**Comment by phadej** [comment]
I don't remember the exact application, but the need is generating TH names without using TemplateHaskell, i.e. the same we use `CURRENT_PACKAGE_KEY` except that one is for current package only: we cannot create `Name`s for things in other packages.

That said, there **no** problem with recompliation (or rather, adding unit ids of dependencies won't make it any worse), as current package key contains the hash (of everything) already, e.g:
```
#ifndef CURRENT_PACKAGE_KEY
#define CURRENT_PACKAGE_KEY "aeson-1.5.6.0-4eed39ada2195fa9448712188bd2ac62f132730beb60cffcb431690177bebc36"
```

(when it's compiled into the store, for local pakcages the unit id is just `...-inplace`)

**Comment by RyanGlScott** [comment]
Thank you, your memory is better than mine in this case. Also, that's a good observation about `CURRENT_PACKAGE_KEY`.
